src/pybind/mgr/prometheus/module.py

```diff
@@ -481,7 +481,7 @@ def _setup_static_metrics(self) -> Dict[str, Metric]:
         for state in DF_POOL:
             path = 'pool_{}'.format(state)
             metrics[path] = Metric(
-                'gauge',
+                'gauge' if state not in ('rd', 'rd_bytes', 'wr', 'wr_bytes') else 'counter',
```

Comment:
Just two questions (out of curiosity) from my side:

1. Would the following line work as well? - It's a bit easier to read to me.

```suggestion
                'counter' if state in ('rd', 'rd_bytes', 'wr', 'wr_bytes') else 'gauge',
```

2. If I understand the tracker issue correctly, the difference between "gauge" and "counter" is that "counter" can't be decreased while "gauge" can. I'm wondering why we're going to use "counter" for reads/read bytes and writes/write bytes, because I assume these values to go up and down over the time. Or do I misunderstand the values here?

Reply:
1. Yes, that would work as well.
2. Yes, a counter only ever goes up whereas the values for a gauge can go up and down. It seems the values for `rd`, `rd_bytes`, `rw` and `rw_bytes` do not ever go down, just up. It also appears that they may be reset at some point or due to some operation, but Prometheus will, due to having the correct type set here, do the right thing and keep adding what is being returned to what Prometheus already has collected. [You can read about Prometheus metric types in more detail on the Prometheus docs about metric types.](https://prometheus.io/docs/concepts/metric_types/)

Reply:
To expand on that a little further, the rate of `rd_bytes` and `wr_bytes` is being requested in the PromQL query of the pool-overview.json ([here](https://github.com/ceph/ceph/blob/master/monitoring/grafana/dashboards/pool-overview.json#L1167) and [here](https://github.com/ceph/ceph/blob/master/monitoring/grafana/dashboards/pool-overview.json#L1333)) and pool-detail.json ([here](https://github.com/ceph/ceph/blob/master/monitoring/grafana/dashboards/pool-detail.json#L444) and [here](https://github.com/ceph/ceph/blob/master/monitoring/grafana/dashboards/pool-detail.json#L451)) Grafana dashboards, respectively. Prometheus then calculates the rate based on the data of the last 30 seconds (in case of pool-overview.json) and based on the data of the last 60 seconds (in case of pool-detail.json), respectively and returns these rates. If those metrics were rates already, it would be wrong to do so and we'd need to fix those PromQL queries in these Grafana panels instead.

Reply:
Thank you for the detailed explanation, @p-se :+1:

Reply:
i concur with @s0nea -- to prefer positive logic over native one for better readability.